cluster.go

```diff
@@ -727,7 +727,14 @@ func (c *Cluster) ID(ctx context.Context) *api.ID {
 	for _, addr := range c.host.Addrs() {
 		addrsSet[addr.String()] = struct{}{}
 	}
+
+	var addrSorted []string
```

Comment:
I am not sure why we are "filtering dups" above. This is very old code. Was `host.Addrs()` producing duplicates?

If we removed that, would we need sorting? Probably the sorting would not flap in that case.

And can we use the canonical https://godoc.org/github.com/libp2p/go-libp2p-core/peer#AddrInfoToP2pAddrs and deprecate `MustLibp2pMultiaddrJoin` thing?

Reply:
There is code to dedup addresses in libp2p: https://github.com/libp2p/go-libp2p/blob/master/p2p/host/basic/basic_host.go#L592

Reply:
(and it does not list from the map so the order is certainly kept)